unittests/rpc/handlers/AccountChannelsTest.cpp

```diff
@@ -0,0 +1,915 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2023, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <rpc/common/AnyHandler.h>
+#include <rpc/ngHandlers/AccountChannels.h>
+#include <util/Fixtures.h>
+#include <util/TestObject.h>
+
+#include <fmt/core.h>
+
+using namespace RPCng;
+namespace json = boost::json;
+using namespace testing;
+
+constexpr static auto LEDGERHASH =
+    "4BC50C9B0D8515D3EAAE1E74B29A95804346C491EE1A95BF25E4AAB854A6A652";
+constexpr static auto ACCOUNT = "rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn";
+constexpr static auto ACCOUNT2 = "rLEsXccBGNR3UPuPu2hUXPjziKC3qKSBun";
+constexpr static auto ACCOUNT3 = "rB9BMzh27F3Q6a5FtGPDayQoCCEdiRdqcK";
+constexpr static auto INDEX1 =
+    "E6DBAFC99223B42257915A63DFC6B0C032D4070F9A574B255AD97466726FC321";
+constexpr static auto INDEX2 =
+    "E6DBAFC99223B42257915A63DFC6B0C032D4070F9A574B255AD97466726FC322";
+constexpr static auto TXNID =
+    "05FB0EB4B899F056FA095537C5817163801F544BAFCEA39C995D76DB4D16F9DD";
+
+class RPCAccountHandlerTest : public SyncAsioContextTest, public MockBackendTest
+{
+    void
+    SetUp() override
+    {
+        SyncAsioContextTest::SetUp();
+        MockBackendTest::SetUp();
+    }
+    void
+    TearDown() override
+    {
+        MockBackendTest::TearDown();
+        SyncAsioContextTest::TearDown();
+    }
+};
+
+TEST_F(RPCAccountHandlerTest, NonHexLedgerHash)
+{
+    boost::asio::spawn(ctx, [this](boost::asio::yield_context yield) {
+        auto const handler =
+            AnyHandler{AccountChannelsHandler{yield, mockBackendPtr}};
```

Comment:
Oops, this `yield` here is **bad**.. imagine this situation, we will have a `std::unordered_map<string, AnyHandler>` of already existing handlers and we will call `process` on the existing handler instances, not create new ones for each response. This means that `yield` will be new each time (and hopefully eventually there will not be one at all). So we can't inject it once for the entire application 😢